torchqdynamics/odeint.py

```diff
@@ -27,6 +50,34 @@ def forward(self, t: float, y: Tensor) -> Tensor:
         """
         pass
 
+    def __init__(self, H: Union[Tensor, Callable]):
+        """
+        Args:
+            H: Hamiltonian, of shape `(b_H, n, n)`.
+        """
+        if isinstance(H, Tensor):
+            if H.ndim == 3:
+                H = H[:, None, ...]  # (b_H, 1, n, n)
+        self._H = H
+        self._cache = {}
+
+    @H_dependent
+    def H(self, t: float) -> Tensor:
+        if isinstance(self._H, Tensor):
+            return self._H
+        elif callable(self._H):
+            return self._H(t)
```

Comment:
I think the batching of H is not handled if it's a callable. So typically if the user sends a time-dependent 2D tensor, it should be converted somewhere to a 4D tensor of size (1, 1, n, n). Just like we do in the time-independent case.